newsuk/times-components
# feat(TDP-1712): remove react-native foreverrrrr

## Description

This is the last stage of the React Native removal work https://nidigitalsolutions.jira.com/browse/TDP-1688. 
[TDP-1712](https://nidigitalsolutions.jira.com/browse/TDP-1712)
[TDP-1712](https://nidigitalsolutions.jira.com/browse/TDP-2320)

The original scope for this work was to remove React Native from the SSR package, but to be able to do so  we had to update the foundations of the repo as it was still very reliant on React Native dependencies and would cause issues if you tried to update SSR alone. Hence how this PR became removing ALL remaining React Native.
This involved updates and changes such as:

- Updating webpack configs
- Updating Babel configs
- Creating a new web friendly jest-configurator package (jest-configurator-web) which creates a reusable jest config. Or for those packages that had more specific local jest configs - we updated those. Then deleted the old jest-configurator package.
- Removed any remaining React Native code and dependencies. This led to removing some redundant tests or behaviour also.
- SSR package -  Updated to use ReactDOM to create the server & client. - Removing React Native's AppRegistry. 
- Updated the ReadME's to no longer include React Native references.
- Updated package.json to remove outdated dependencies & devDependencies

## To test
- Remove all node_modules and dist folders. `rm -rf node_modules dist`
- Run `yarn build:local` - This will yarn install and bundle etc.
- Then run all the test commands from the root `yarn test:web`, `yarn test:unit`, `yarn test:e2e`, `yarn test:all`
- Run storybook and check it is working locally.

Note - In the Card package you might get snapshot errors when you run test:web - This is due to styled component classnames failing, but they will fail in circle if you update them locally. So if you have this issue you can just ignore them and run `yarn test:web --no-bail` 

Have fuuuunnnn!
